io/src/main/scala/fs2/io/file/file.scala

```diff
@@ -119,4 +122,81 @@ package object file {
+    * Creates a stream of [[Path]]s inside a directory.
+    */
+  def streamDirectory[F[_]: Sync: ContextShift](blocker: Blocker, path: Path): Stream[F, Path] =
+    _runJavaCollectionResource[F, DirectoryStream[Path]](
+      blocker,
+      blocker.delay(Files.newDirectoryStream(path)),
+      _.asScala.iterator)
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory, filtering the results by the given predicate.
+    */
+  def streamDirectory[F[_]: Sync: ContextShift](blocker: Blocker,
+                                                path: Path,
+                                                filter: Path => Boolean): Stream[F, Path] =
+    _runJavaCollectionResource[F, DirectoryStream[Path]](
+      blocker,
+      blocker.delay(Files.newDirectoryStream(path, (entry: Path) => filter(entry))),
+      _.asScala.iterator)
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory which match the given glob.
+    */
+  def streamDirectory[F[_]: Sync: ContextShift](blocker: Blocker,
+                                                path: Path,
+                                                glob: String): Stream[F, Path] =
+    _runJavaCollectionResource[F, DirectoryStream[Path]](
+      blocker,
+      blocker.delay(Files.newDirectoryStream(path, glob)),
+      _.asScala.iterator)
+
+  /**
+    * Creates a stream of [[Path]]s contained in a given file tree. Depth is unlimited.
+    */
+  def walk[F[_]: Sync: ContextShift](blocker: Blocker, start: Path): Stream[F, Path] =
+    walk[F](blocker, start, Seq.empty)
+
+  /**
+    * Creates a stream of [[Path]]s contained in a given file tree, respecting the supplied options. Depth is unlimited.
+    */
+  def walk[F[_]: Sync: ContextShift](blocker: Blocker,
+                                     start: Path,
+                                     options: Seq[FileVisitOption]): Stream[F, Path] =
+    walk[F](blocker, start, Int.MaxValue, options)
+
+  /**
+    * Creates a stream of [[Path]]s contained in a given file tree down to a given depth.
+    */
+  def walk[F[_]: Sync: ContextShift](blocker: Blocker,
+                                     start: Path,
+                                     maxDepth: Int,
+                                     options: Seq[FileVisitOption] = Seq.empty): Stream[F, Path] =
+    _runJavaCollectionResource[F, JStream[Path]](
+      blocker,
+      blocker.delay(Files.walk(start, maxDepth, options: _*)),
+      _.iterator.asScala)
+
+  private def _runJavaCollectionResource[F[_]: Sync: ContextShift, C <: AutoCloseable](
+      blocker: Blocker,
+      javaCollection: F[C],
+      collectionIterator: C => Iterator[Path]): Stream[F, Path] =
+    Stream
+      .resource(Resource.fromAutoCloseable(javaCollection))
+      .flatMap(ds => _runIteratorOnBlocker[F, Path](blocker, collectionIterator(ds)))
+
+  // Like Stream.fromIterator, but on the given Blocker
+  private def _runIteratorOnBlocker[F[_]: ContextShift, A](blocker: Blocker, iterator: Iterator[A])(
```

Comment:
Thoughts on making this `Stream.fromBlockingIterator`?

Reply:
Done